File: Straysafe.Frontend/src/components/utilities/models/chatUtility/ChatData.js
import { ChatPosition } from "./ChatPosition";

export class ChatData{
    constructor({chatInfo, sender, recepient, message, type}) {
        this.id = 0;
        this.chatInfo = chatInfo;
        this.sender = sender;
        this.recepient = recepient;
        this.message = message;
        this.type = type;
        this.messageDate = new Date();

        this.Position = ChatPosition.Left;

    }
}

File: Straysafe.Backend/Common/DAL/Models/ChatData.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Straysafe.Backend.Common.DAL.Models
{
    public class ChatData
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public Guid ChatInfo { get; set; }
        public Guid Sender { get; set; }
        public Guid Recepient { get; set; }
        public string Message { get; set; } = string.Empty;
        public string Type { get; set; } = "msg"; /* msg | attachment */
        public DateTime MessageDate { get; set; } = DateTime.Now;
    }
}
